chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -72,36 +234,48 @@ void MdTextButtonBase::UpdateColorsForBrave() {
   }
 }
 
-MdTextButton::MdTextButton(PressedCallback callback,
-                           const std::u16string& text,
-                           int button_context)
-    : MdTextButtonBase(std::move(callback), text, button_context) {
-  SetCornerRadius(100);
-  views::HighlightPathGenerator::Install(
-      this, std::make_unique<BraveTextButtonHighlightPathGenerator>());
-  auto* ink_drop = views::InkDrop::Get(this);
-  views::InkDrop::UseInkDropForFloodFillRipple(ink_drop,
-                                               /*highlight_on_hover=*/false,
-                                               /*highlight_on_focus=*/true);
-  ink_drop->SetCreateHighlightCallback(base::BindRepeating(
-      [](Button* host) {
-        const SkColor fill_color = SK_ColorTRANSPARENT;
-        gfx::RectF boundsF(host->GetLocalBounds());
-        return std::make_unique<InkDropHighlight>(
-            boundsF.size(), static_cast<MdTextButton*>(host)->GetCornerRadius(),
-            boundsF.CenterPoint(), fill_color);
-      },
-      this));
-}
+// To be called from MdTextButtonBase::UpdateColors().
+void MdTextButton::UpdateColorsForBrave() {
+  if (GetKind() == Kind::OLD) {
+    UpdateOldColorsForBrave();
+    return;
+  }
 
-MdTextButton::~MdTextButton() = default;
+  // Theme should only ever be |absl::nullopt| if the button is Prominent.
+  DCHECK(theme_);
 
-SkPath MdTextButton::GetHighlightPath() const {
-  SkPath path;
-  int radius = GetCornerRadius();
-  path.addRRect(
-      SkRRect::MakeRectXY(RectToSkRect(GetLocalBounds()), radius, radius));
-  return path;
+  absl::optional<ButtonStyle> style;
+  auto is_dark = GetNativeTheme()->GetPreferredColorScheme() ==
+                 ui::NativeTheme::PreferredColorScheme::kDark;
+  auto state = GetVisualState();
+  if (theme_) {
+    style = is_dark ? theme_->normal_dark : theme_->normal_light;
+    if (state == STATE_HOVERED || state == STATE_PRESSED)
+      style = is_dark ? theme_->hover_dark : theme_->hover_light;
+    if (GetLoading())
+      style = is_dark ? theme_->loading_dark : theme_->loading_light;
+    if (!GetEnabled() || state == STATE_DISABLED)
+      style = is_dark ? theme_->disabled_dark : theme_->disabled_light;
```

Comment:
b4. then we might be able to have code here, 

```
const auto style = theme_.at(kDark).at(GetVisualState);  // using at() will complain if there's no value for the key.
```

Reply:
Updated - I also collapsed the loading/disabled states into an opacity override on the `STATE_NORMAL`, which is what the are in the design doc :smile: